src/mapping/nav2_map_server/test/unit/test_occ_grid.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2008, Willow Garage, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* Author: Brian Gerkey */
+
+#include <gtest/gtest.h>
+#include <experimental/filesystem>
+#include <stdexcept>
+#include <string>
+
+#include "nav2_map_server/map_reps/occ_grid_server.hpp"
+#include "test_constants/test_constants.h"
+
+#define TEST_DIR TEST_DIRECTORY
+
+using namespace std; // NOLINT
+using std::experimental::filesystem::path;
+
+class OccGridTest : public nav2_map_server::OccGridServer
+{
+public:
+  OccGridTest() {}
+  ~OccGridTest() {}
+
+  nav_msgs::srv::GetMap::Response GetMap() {return occ_resp_;}
+};
+
+class MapServerTest : public ::testing::Test
+{
+protected:
+  OccGridTest * OccTest;
+  nav_msgs::srv::GetMap::Response map_resp;
+  MapServerTest()
```

Comment:
Did you mean for the constructor to be protected?

Reply:
I was following an example where the test fixture had a protected constructor. I suppose it could be public.